foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-APIManagementPolicySetDefinition.json

```diff
@@ -84,6 +84,328 @@
+                                "metadata": {
+                                    "displayName": "Effect",
+                                    "description": "Enable or disable the execution of the policy"
+                                },
+                                "allowedValues": [
+                                    "Audit",
+                                    "Deny",
+                                    "Disabled"
+                                ],
+                                "defaultValue": "Deny"
+                            },
+                            "evaluatedSkuNames": {
+                                "type": "Array",
+                                "metadata": {
+                                    "displayName": "API Management SKU Names",
+                                    "description": "List of API Management SKUs against which this policy will be evaluated."
+                                },
+                                "allowedValues": [
+                                    "Developer",
+                                    "Basic",
+                                    "Standard",
+                                    "Premium",
+                                    "Consumption"
+                                ],
+                                "defaultValue": [
+                                    "Developer",
+                                    "Premium"
+                                ]
+                            },
+                            "virtualNetworkType": {
+                                "type": "String",
+                                "metadata": {
+                                    "displayName": "Virtual Network Type",
+                                    "description": "Specifies the desired virtual network type that should be enforced."
+                                },
+                                "allowedValues": [
+                                    "None",
+                                    "Internal",
+                                    "External"
+                                ],
+                                "defaultValue": "Internal"
+                            }
+                        },
+                        "policyRule": {
+                            "if": {
+                                "allOf": [
+                                    {
+                                        "field": "type",
+                                        "equals": "Microsoft.ApiManagement/service"
+                                    },
+                                    {
+                                        "field": "Microsoft.ApiManagement/service/sku.name",
+                                        "in": "[[parameters('evaluatedSkuNames')]"
+                                    },
+                                    {
+                                        "anyOf": [
+                                            {
```

Comment:
How would this work with the custom policy we have already included ("API Management services should use a virtual network")?

Reply:
This is more restricted and enforces to host the instance in `Internal` mode instead of allowing both `Internal` and `External`. I am assuming that by default the APIM instances in Corp should not be accessible from the internet? See link [here](https://learn.microsoft.com/en-us/azure/api-management/virtual-network-concepts?tabs=stv2#access-options).

Reply:
Do we have to distinguish at this level? Remember that this policySetDefinition will be assigned broadly at the LZ scope and apply to both cloud-native and corp connected. @victorar - any input from your PoV wrt to the other policies we have for APIM related to networking?

Reply:
I will follow your guidance and understand the concerns.
If we enforce this for corp and online Landing Zones, we should still be fine. Corp use-cases will then be able to rely on private endpoints whereas online landing zones can use an App GW with WAF and Public IP to accept public traffic.
Looking forward to your input @victorar and will remove it if it is too restrictive.

Reply:
Please let me know how to continue here.